test/filters/test_crop_hull.cpp

```diff
@@ -0,0 +1,495 @@
+    {
+      for (const float i: {0.f, 1.f})
+        for (const float j: {0.f, 1.f})
+          for (const float k: {0.f, 1.f})
+            baseInputCloud->emplace_back(i, j, k);
+    }
+    protected:
+      pcl::PointCloud<pcl::PointXYZ>::ConstPtr getBaseInputCloud() const override {
+        return baseInputCloud;
+      }
+      int getDim() const override {
+        return 3;
+      }
+    private:
+      pcl::PointCloud<pcl::PointXYZ>::Ptr baseInputCloud;
+  };
+}
+
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// since test input cloud has same distribution for all dimensions, this test also check problem from issue #3960 //
+TEST_F (PCLCropHullTestFixture2d, 2dsquare_simple_test)
+{
+  auto checker = [](pcl::CropHull<pcl::PointXYZ> cropHullFilter, TestData const & testData) {
+    cropHullFilter.setInputCloud(testData.inputCloud);
+    pcl::Indices filteredIndices;
+    cropHullFilter.filter(filteredIndices);
+    pcl::test::EXPECT_EQ_VECTORS(testData.insideIndices, filteredIndices);
+  };
+  test(checker);
+}
+
+//////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// this test will fail //
+//TEST_F (PCLCropHullTestFixture2d, 2dsquare_test_reentrancy)
+//{
+//  auto checker = [](pcl::CropHull<pcl::PointXYZ> cropHullFilter, TestData const & testData) {
+//    pcl::Indices filteredIndices;
+//    cropHullFilter.filter(filteredIndices);
+//    cropHullFilter.filter(filteredIndices);
+//    pcl::test::EXPECT_EQ_VECTORS(testData.insideIndices, filteredIndices);
+//  };
+//  test(checker);
+//}
+
+//////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// this test will fail //
+//TEST_F (PCLCropHullTestFixture2d, 2dsquare_test_negative_filter_functionality)
+//{
+//  auto checker = [](pcl::CropHull<pcl::PointXYZ> cropHullFilter, TestData const & testData) {
+//    cropHullFilter.setNegative(true);
+//    pcl::Indices filteredIndices;
+//    cropHullFilter.filter(filteredIndices);
+//    pcl::test::EXPECT_EQ_VECTORS(testData.outsideIndices, filteredIndices);
+//    pcl::test::EXPECT_EQ_VECTORS(testData.insideIndices, *cropHullFilter.getRemovedIndices());
+//  };
+//  test(checker);
+//}
+
+//////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+TEST_F (PCLCropHullTestFixture2d, 2dsquare_test_crop_inside)
+{
+  auto checker = [](pcl::CropHull<pcl::PointXYZ> cropHullFilter, TestData const & testData) {
+    cropHullFilter.setCropOutside(false);
+    pcl::Indices filteredIndices;
+    cropHullFilter.filter(filteredIndices);
+    pcl::test::EXPECT_EQ_VECTORS(testData.outsideIndices, filteredIndices);
+  };
+  test(checker);
+}
+
+//////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// this test will fail //
```

Comment:
Delete all tests that will fail, but are part of the parent PR

Reply:
applied suggestion